extra/src/main/java/aima/extra/probability/DoubleProbabilityNumber.java

```diff
@@ -309,7 +244,96 @@ public boolean sumsToOne(Iterable<ProbabilityNumber> allProbabilities) {
 		return this.isOne();
 	}
 
+	/**
+	 * Override the precision of ProbabilityNumber instances returned as a
+	 * result of performing operations.
+	 * 
+	 * @param mc
+	 */
+	@Override
+	public void overrideComputationPrecisionGlobally(MathContext mc) {
+		if (mc.getPrecision() > 15) {
+			throw new IllegalArgumentException("Maximum precision possible for DoubleProbabilityNumber is 15");
+		}
+		MAX_PRECISION = mc.getPrecision();
+		ROUNDING_MODE = mc.getRoundingMode();
+	}
+
+	/**
+	 * Checks if argument implementing ProbabilityNumber interface is equal to
+	 * the value of the current DoubleProbabilityNumber.
+	 * 
+	 * @param that
+	 *            the ProbabilityNumber type that is to be compared with this
+	 *            DoubleProbabilityNumber.
+	 *
+	 * @return true if this == that, false otherwise
+	 */
+	@Override
+	public boolean equals(Object that) {
+		DoubleProbabilityNumber second = toInternalType((ProbabilityNumber) that);
```

Comment:
You may want to do a conditional check on the type here:

if (that instanceof ProbabilityNumber)
...
} else {
return false;